DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/gzip.cpp

```diff
@@ -154,6 +156,10 @@ int main(int argc, char *argv[]) {
 
 #ifdef FPGA_EMULATOR
     CompressFile(q, infilename, outfilenames, 1, true);
+#elif FPGA_SIMULATOR
+    // warmup run
+    CompressFile(q, infilename, outfilenames, 1, false);
```

Comment:
Why would you need a warmup run when running on the simulator?

Reply:
I would assume that simulator also has some first kernel setups similar to hardware. But I just realized that I'm also skipping the part where performance is measured in simulation, so regardless a warmup run is not needed :)